mesonbuild/interpreterbase.py

```diff
@@ -404,18 +458,26 @@ def method_call(self, node):
         if isinstance(obj, int):
             return self.int_method_call(obj, method_name, args)
         if isinstance(obj, list):
-            return self.array_method_call(obj, method_name, self.reduce_arguments(args)[0])
+            return self.array_method_call(obj, method_name, args)
```

Comment:
Seems unrelated?

Reply:
I made these all behave the same. There's no reason to keep the inconsistency and doing a full MR for this change seems excessive.